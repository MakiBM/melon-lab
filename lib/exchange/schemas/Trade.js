// @flow
import BigNumber from "bignumber.js";

import type { BuyOrSell } from "./BuyOrSell";

/**
 * An executed trade == a taken order
 */
export type Trade = {
  maker: string,
  taker: string,
  timeStamp: Date,
  price: BigNumber,
  type: BuyOrSell,
  quantity: BigNumber,
};
