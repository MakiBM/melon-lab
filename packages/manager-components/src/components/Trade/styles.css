.trade {
  width: 300px;
  margin: var(--space-l) 0;
}
