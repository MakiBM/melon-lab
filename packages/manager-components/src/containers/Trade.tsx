import { withState } from 'recompose';
import compose from 'recompose/compose';

const withActiveTabState = withState('activeTabIndex', 'setTabIndex', 0);

export default compose(withActiveTabState);
