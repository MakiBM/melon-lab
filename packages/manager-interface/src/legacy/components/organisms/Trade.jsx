import React from 'react';
import { Field } from 'redux-form';
import { List, Button, Card, Image, Menu } from 'semantic-ui-react';

import NumberInput from '../molecules/NumberInput';
import displayNumber from '../../utils/displayNumber';

const orderStrategySelector = ({ input }) => (
  <div>
    <Menu text style={{ display: 'flex', justifyContent: 'center' }}>
      <Menu.Item
        name="Market"
        active={input.value === 'Market'}
        onClick={() => input.onChange('Market')}
      />
      <div style={{ marginTop: '0.7em' }}>|</div>
      <Menu.Item
        name="Limit"
        active={input.value === 'Limit'}
        onClick={() => input.onChange('Limit')}
      />
    </Menu>
  </div>
);

const orderTypeSelector = ({
  input,
  baseTokenSymbol,
  quoteTokenSymbol,
  theirOrderType,
  disabled,
}) => (
  <button
    id="switchButton"
    disabled={disabled}
    style={{ textAlign: 'center', cursor: 'pointer' }}
    onClick={event => {
      event.preventDefault();
      input.onChange(theirOrderType);
    }}
  >
    <Card.Meta>
      {input.value} <strong>{baseTokenSymbol}</strong>
    </Card.Meta>
    <Image
      src="./static/switch.svg"
      width="18em"
      centered
      style={{ cursor: 'pointer' }}
    />

    <Card.Meta>
      {theirOrderType} <strong>{quoteTokenSymbol}</strong>
    </Card.Meta>
  </button>
);

const Trade = ({
  orderType,
  theirOrderType,
  baseTokenSymbol,
  quoteTokenSymbol,
  handleSubmit,
  loading,
  strategy,
  selectedOrder,
  dataValid,
}) => (
  <Card centered id="trade">
    <Card.Content>
      <Card.Header>Trade</Card.Header>
      <form onSubmit={handleSubmit}>
        <Field
          name="type"
          component={orderTypeSelector}
          baseTokenSymbol={baseTokenSymbol}
          quoteTokenSymbol={quoteTokenSymbol}
          theirOrderType={theirOrderType}
          disabled={strategy === 'Market'}
        />
        <br />
        <Field name="strategy" component={orderStrategySelector} />
        {strategy === 'Market' ? <p>Select from the orderbook</p> : null}
        <List>
          <List.Item>
            <List.Content>
              <Field
                id="trade-price"
                disabled={strategy === 'Market' || !dataValid}
                format={displayNumber}
                name="price"
                component={NumberInput}
                label="Price"
              />
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content>
              <Field
                id="trade-quantity"
                disabled={
                  (strategy === 'Market' && !selectedOrder) || !dataValid
                }
                format={displayNumber}
                name="quantity"
                component={NumberInput}
                label="Quantity"
              />
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content>
              <Field
                id="trade-total"
                disabled={
                  (strategy === 'Market' && !selectedOrder) || !dataValid
                }
                format={displayNumber}
                name="total"
                component={NumberInput}
                label="Total"
              />
            </List.Content>
          </List.Item>
        </List>

        {!dataValid ? (
          <p style={{ color: 'rgb(209, 102, 102)' }}>
            Trading not possible when price feed down
          </p>
        ) : null}

        <Button
          basic
          id="tradeButton"
          color="black"
          style={{ width: '100%' }}
          disabled={(strategy === 'Market' && !selectedOrder) || !dataValid}
        >
          {orderType}
        </Button>
      </form>
    </Card.Content>
    <div className={`ui ${loading ? 'active' : ''} inverted dimmer`}>
      <div className="ui text loader">Executing your order ...</div>
    </div>
  </Card>
);

export default Trade;
