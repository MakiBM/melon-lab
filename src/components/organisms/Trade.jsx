import React from "react";
import { Field } from "redux-form";
import { List, Button, Card, Image } from "semantic-ui-react";

import renderInput from "../utils/renderInput";

const Trade = ({
  switchSymbols,
  orderType,
  theirOrderType,
  baseTokenSymbol,
  quoteTokenSymbol,
  handleSubmit,
  loading,
  onChange,
}) => (
  <form onSubmit={handleSubmit}>
    <Card centered id="trade">
      <Card.Content>
        <Card.Header>Place an order</Card.Header>
        <br />
        <br />

        <button
          style={{ textAlign: "center", cursor: "pointer" }}
          onClick={switchSymbols}
        >
          <Card.Meta>
            {orderType} <strong>{baseTokenSymbol}</strong>
          </Card.Meta>
          <Image
            src="./switch.svg"
            width="18em"
            centered
            style={{ cursor: "pointer" }}
          />

          <Card.Meta>
            {theirOrderType} <strong>{quoteTokenSymbol}</strong>
          </Card.Meta>
        </button>

        <br />
        <List>
          <List.Item as="a">
            <List.Content>
              <Field
                name="price"
                component={renderInput}
                onChange={onChange}
                label="Price"
              />
            </List.Content>
          </List.Item>
          <List.Item as="a">
            <List.Content>
              <Field
                name="quantity"
                component={renderInput}
                onChange={onChange}
                label="Quantity"
              />
            </List.Content>
          </List.Item>
          <List.Item as="a">
            <List.Content>
              <Field
                name="total"
                component={renderInput}
                onChange={onChange}
                label="Total"
              />
            </List.Content>
          </List.Item>
        </List>

        <Button basic color="black" style={{ width: "100%" }}>
          {orderType}
        </Button>
      </Card.Content>
      <div className={`ui ${loading ? "active" : ""} inverted dimmer`}>
        <div className="ui text loader">Executing your order ...</div>
      </div>
    </Card>
  </form>
);

export default Trade;
