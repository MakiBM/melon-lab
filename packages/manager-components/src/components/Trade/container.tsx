import { withState } from 'recompose';
import compose from 'recompose/compose';
import Trade from './index';

const withActiveTabState = withState('activeTabIndex', 'setTabIndex', 0);

export default compose(withActiveTabState)(Trade);
