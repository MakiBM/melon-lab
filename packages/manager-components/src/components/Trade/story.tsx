import { storiesOf } from '@storybook/react';
import React from 'react';
import Trade from './index';

storiesOf('Components|Trade', module).add('Default', () => {
  return <Trade />;
});
