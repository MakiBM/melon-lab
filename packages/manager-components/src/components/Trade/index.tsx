import React, { StatelessComponent } from 'react';
import { TabContent, Tabs } from '~/blocks/Tabs';
import OrderForm from '~/components/OrderForm';
import withActiveState from '~/containers/Trade';

import styles from './styles.css';

export interface OrderFormProps {
  activeTabIndex: string;
  setTabIndex(index: number);
}

export const Trade: StatelessComponent<OrderFormProps> = ({
  activeTabIndex,
  setTabIndex,
}) => {
  return (
    <Tabs
      handleTabClick={index => setTabIndex(index)}
      activeTabIndex={activeTabIndex}
    >
      <style jsx>{styles}</style>
      <TabContent title="Take">
        <OrderForm />
      </TabContent>
      <TabContent title="Place">
        <OrderForm />
      </TabContent>
    </Tabs>
  );
};

export default (withActiveState)(Trade);
