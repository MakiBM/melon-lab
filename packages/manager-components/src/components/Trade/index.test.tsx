import React from 'react';
import Trade from './index';

describe('Trade', () => {
  const defaultElement = <Trade />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
