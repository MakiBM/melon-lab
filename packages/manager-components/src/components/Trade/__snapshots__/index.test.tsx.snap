// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trade should render correctly 1`] = `
<Component
  activeTabIndex={0}
  setTabIndex={[Function]}
/>
`;
